### PR TITLE
CPM-134: add roles export

### PR DESCRIPTION
**Pull request by nmarniesse**

### CODE REVIEW

File: src/Akeneo/UserManagement/Component/Connector/ArrayConverter/StandardToFlat/Role.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    {
+        foreach ($this->aclManager->getAllExtensions() as $extension) {
+            if ($extension->getExtensionKey() === $type) {
+                return $extension->getPermissions();
+            }
+        }
+
+        return [];
+    }
+}
```
Comment (mmetayer) [inline src/Akeneo/UserManagement/Component/Connector/ArrayConverter/StandardToFlat/Role.php:135]: IMO we don't need this; it's just an array converter, we should already have all the information needed.
I'd go for a simple 
```php
return implode(',', array_map(
    fn (array $privilege) => $privilege['id'],
    $privileges
));
```
it's enough for our current use-case. We can still create another implem in the future if we want to handle a more specific use case

Comment (nmarniesse) [reply src/Akeneo/UserManagement/Component/Connector/ArrayConverter/StandardToFlat/Role.php]: Fixed

File: src/Akeneo/UserManagement/Component/Normalizer/RoleNormalizer.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+/**
+ * @author    Nicolas Marniesse <nicolas.marniesse@akeneo.com>
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+final class RoleNormalizer implements NormalizerInterface, CacheableSupportsMethodInterface
+{
+    private const ACL_EXTENSION_KEY = 'action';
+
+    protected array $supportedFormats = ['array', 'standard'];
```
Comment (mmetayer) [inline src/Akeneo/UserManagement/Component/Normalizer/RoleNormalizer.php:22]: private :)

File: src/Oro/Bundle/SecurityBundle/Normalizer/AclPrivilegeNormalizer.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+use Webmozart\Assert\Assert;
+
+/**
+ * @author    Nicolas Marniesse <nicolas.marniesse@akeneo.com>
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+final class AclPrivilegeNormalizer implements NormalizerInterface, CacheableSupportsMethodInterface
+{
+    protected array $supportedFormats = ['array', 'standard'];
```
Comment (mmetayer) [inline src/Oro/Bundle/SecurityBundle/Normalizer/AclPrivilegeNormalizer.php:19]: private

File: tests/back/UserManagement/Integration/Bundle/Export/ExportUserRoleIntegration.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+use Akeneo\Test\Integration\TestCase;
+use Akeneo\Test\IntegrationTestsBundle\Launcher\JobLauncher;
+use Akeneo\Tool\Bundle\BatchBundle\Persistence\Sql\SqlCreateJobInstance;
+
+final class ExportUserRoleIntegration extends TestCase
+{
+    private const CSV_EXPORT_JOB_CODE = 'csv_user_role_export';
+    private const XLSX_EXPORT_JOB_CODE = 'xlsx_user_role_export';
+
+    protected JobLauncher $jobLauncher;
```
Comment (mmetayer) [inline tests/back/UserManagement/Integration/Bundle/Export/ExportUserRoleIntegration.php:15]: private

**Review by mmetayer — APPROVED** [review]
:ok_hand:

**Pull request merged**